File: files/UsuarioService.java
// Exemplo de código Java com violações LGPD
// Este arquivo é usado para demonstrar o LGPD Guard

package com.exemplo.usuario;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/usuarios")
public class UsuarioService {

    private static final Logger log = LoggerFactory.getLogger(UsuarioService.class);

    // ❌ VIOLAÇÃO Art. 46: CPF em log sem mascaramento
    public void criarUsuario(Usuario usuario) {
        log.info("Criando usuário com CPF: " + usuario.getCpf());

        // ❌ VIOLAÇÃO Art. 46: senha em texto puro
        String senha = usuario.getSenha();
        database.save("INSERT INTO usuarios (cpf, senha) VALUES (?, ?)",
                      usuario.getCpf(), senha);

        // ❌ VIOLAÇÃO Art. 6 (finalidade): CPF enviado para analytics
        analyticsClient.track("user_created", Map.of(
            "cpf", usuario.getCpf(),
            "email", usuario.getEmail()
        ));
    }

    // ❌ VIOLAÇÃO Art. 46: comunicação sem HTTPS
    public void sincronizarDados(Usuario usuario) {
        String url = "http://servico-externo.com.br/api/dados";
        httpClient.post(url, usuario);
    }

    // ✅ CORRETO: CPF mascarado no log
    public void atualizarUsuarioCorreto(Usuario usuario) {
        String cpfMascarado = "***." + usuario.getCpf().substring(4, 7) + ".***-**";
        log.info("Atualizando usuário CPF: " + cpfMascarado);

        // ✅ CORRETO: senha com hash BCrypt
        String senhaHash = bcryptEncoder.encode(usuario.getSenha());
        database.save("UPDATE usuarios SET senha = ? WHERE id = ?",
                      senhaHash, usuario.getId());
    }
}

File: examples/violations/usuario_service.py
# Exemplo de código Python com violações LGPD
# Este arquivo é usado para demonstrar o LGPD Guard

import logging
import requests

logger = logging.getLogger(__name__)


class UsuarioService:

    def criar_usuario(self, usuario: dict):
        # ❌ VIOLAÇÃO Art. 46: CPF exposto em log
        logger.info(f"Criando usuário CPF: {usuario['cpf']}")

        # ❌ VIOLAÇÃO Art. 46: senha em texto puro no banco
        self.db.execute(
            "INSERT INTO usuarios (cpf, email, senha) VALUES (?, ?, ?)",
            (usuario['cpf'], usuario['email'], usuario['senha'])
        )

        # ❌ VIOLAÇÃO Art. 6: dado pessoal enviado para analytics sem base legal
        requests.post("https://analytics.exemplo.com/track", json={
            "event": "user_created",
            "cpf": usuario['cpf'],
            "email": usuario['email']
        })

    def buscar_por_cpf(self, cpf: str):
        # ❌ VIOLAÇÃO Art. 46: SQL injection com dado pessoal
        query = f"SELECT * FROM usuarios WHERE cpf = '{cpf}'"
        return self.db.execute(query)

    # ✅ CORRETO: CPF mascarado
    def buscar_por_cpf_correto(self, cpf: str):
        logger.info(f"Buscando usuário CPF: ***.***.{cpf[-6:-2]}-**")
        return self.db.execute(
            "SELECT * FROM usuarios WHERE cpf = ?", (cpf,)
        )

# teste
logger.info(f'CPF do usuario: {usuario["cpf"]}')
